lib/segment/src/index/field_index/full_text_index/inverted_index.rs

```diff
@@ -0,0 +1,193 @@
+    }
+
+    pub fn remove_document(&mut self, idx: PointOffsetType) -> Option<Document> {
+        if self.point_to_docs.len() <= idx as usize {
+            return None; // Already removed or never actually existed
+        }
+
+        let removed_doc = std::mem::take(&mut self.point_to_docs[idx as usize]);
+
+        if removed_doc.is_empty() {
+            return None;
+        }
+        self.points_count -= 1;
+
+        for removed_token in &removed_doc.tokens {
+            let posting = self.postings.get_mut(removed_token);
+            if let Some(posting) = posting {
+                posting.remove(&idx);
+                if posting.is_empty() {
+                    self.postings.remove(removed_token);
+                }
+            }
+        }
+        Some(removed_doc)
+    }
+
+    pub fn filter(&self, query: &ParsedQuery) -> Box<dyn Iterator<Item = PointOffsetType> + '_> {
+        let postings_opt: Option<Vec<_>> = query
+            .tokens
+            .iter()
+            .map(|token| self.postings.get(token))
+            .collect();
+        if postings_opt.is_none() {
+            // There are unseen tokens -> no matches
+            return Box::new(vec![].into_iter());
+        }
+        let postings = postings_opt.unwrap();
+        if postings.is_empty() {
+            return Box::new(
+                self.point_to_docs
+                    .iter()
+                    .enumerate()
+                    .filter_map(|(idx, doc)| {
+                        if doc.is_empty() {
+                            None
+                        } else {
+                            Some(idx as PointOffsetType)
+                        }
+                    }),
+            );
+        }
+        intersect_btree_iterator(postings)
+    }
+
+    pub fn estimate_cardinality(
+        &self,
+        query: &ParsedQuery,
+        condition: &FieldCondition,
+    ) -> CardinalityEstimation {
+        let postings_opt: Option<Vec<_>> = query
+            .tokens
+            .iter()
+            .map(|token| self.postings.get(token))
+            .collect();
+        if postings_opt.is_none() {
+            // There are unseen tokens -> no matches
+            return CardinalityEstimation {
+                primary_clauses: vec![PrimaryCondition::Condition(condition.clone())],
+                min: 0,
+                exp: 0,
+                max: 0,
+            };
+        }
+        let postings = postings_opt.unwrap();
+        if postings.is_empty() {
+            return CardinalityEstimation::unknown(self.points_count);
+        }
+        // Smallest posting is the largest possible cardinality
+        let smallest_posting = postings.iter().map(|posting| posting.len()).min().unwrap();
+
+        return if postings.len() == 1 {
+            CardinalityEstimation {
+                primary_clauses: vec![PrimaryCondition::Condition(condition.clone())],
+                min: smallest_posting,
```

Comment:
I expected `min: 0` because `small_posting` definition is commented as `Smallest posting is the largest possible cardinality`. Largest possible cardinality != min cardinality.

Reply:
If there is only one posting, largest possible cardinality == min cardinality

Reply:
okay